src/main/java/org/jabref/model/util/OptionalUtil.java

```diff
@@ -41,4 +43,16 @@
     public static <T, R> Stream<R> flatMap(Optional<T> value, Function<? super T, ? extends Collection<? extends R>> mapper) {
         return toStream(value).flatMap(element -> mapper.apply(element).stream());
     }
+
+    public static <T> Boolean isPresentAnd(Optional<T> value, Predicate<T> check) {
```

Comment:
Sometimes the use of optional is a bit inconvenient and redundant but for me these helper methods add additional complexity and twist my head while reading the code.